### ISSUE TITLE
Feature request: warn if DiagnosticAnalyzerAttribute is used on a type that is not a DiagnosticAnalyzer

### ISSUE THREAD

**Issue by JohanLarsson** [dotnet/roslyn-analyzers#1690]
#### Analyzer package

Microsoft.CodeAnalysis.Analyzers

#### Repro steps

```cs
namespace RoslynSandbox
{
    using Microsoft.CodeAnalysis;
    using Microsoft.CodeAnalysis.Diagnostics;

    [DiagnosticAnalyzer(LanguageNames.CSharp)]
    internal static class Foo
    {
    }
}
```

**Comment by Evangelink** [comment]
@mavasani ping

**Comment by JohanLarsson** [comment]
This is actually a pretty useful analyzer. Having the attribute on a class that is not a subclass of `DiagnosticAnalyzer` meant that no analyzer from the .nuget worked in VS.
